src/routes/measurement-threshold/find.ts

```diff
@@ -0,0 +1,110 @@
+import Joi from 'joi';
+import { Edge } from '../../models/edge';
+import { Pin } from '../../models/pin';
+import { ControllerGeneratorOptionsWithClient } from '../../comms/controller';
+
+import { PinGroup } from '../../models/pin-group';
+
+import { TableQuery, EffectiveTableQuery, tableQuerySchemaGenerator } from '../../comms/table-controller';
+import { Quantity } from '../../models/quantity';
+import { stringBeforeV7ElseStringOrTranslationSchema } from '../../models/string-or-translations';
+import { schema as siNumberSchema, SiNumber } from '../../models/si-number';
+
+interface Query extends TableQuery {
+  quantityHashId?: string | null;
+  pinGroupHashId?: string | null;
+  edgeHashId?: string | null;
+  pinHashId?: string | null;
+  type?: 'quantity' | 'port';
+}
+
+type Request = {
+  query?: Query;
+} | undefined;
+
+interface EffectiveQuery extends EffectiveTableQuery {
+    quantityHashId?: string | null;
+    pinGroupHashId?: string | null;
+    edgeHashId?: string | null;
+    pinHashId?: string | null;
+    type: 'quantity' | 'port';
+}
+
+interface EffectiveRequest {
+  query: EffectiveQuery;
+}
+
+interface Threshold {
+  criticallyLow: SiNumber | null;
+  low: SiNumber | null;
+  high: SiNumber | null;
+  criticallyHigh: SiNumber | null;
+}
```

Comment:
Can't you just import src/models/threshold.ts? (And same for the schema below)

Reply:
Do you mean this [one](https://github.com/withthegrid/platform-sdk/pull/355/files#diff-2c5237690fe003707fb8b6741dceb16d49fc7defdf7dbc68f114cd0b6cb73200L16)? Maybe I can extend the MeasurementThreshold from the one above instead

Reply:
That's the one indeed. 

> Maybe I can extend the MeasurementThreshold from the one above instead

I would not derive a model from an interface in a controller, but other than that it would be possible. But I would be fine by using the whole `MeasurementThreshold` (so incl. `hashId` and `pinHashId`) here

Reply:
Sorry, I meant to move it to the model and use it then in the controller. But by using the whole `MeasurementThreshold` would mean I also have to return those extra fields and the `threshold` object might come from a `quantity` and not have a `pinHashId`

Reply:
Ah, fair enough!